**Bioinformatics, genomics, agriculture, food science, medicine, genetic engineering, etc...**

+ [BIOGERONTOLOGY](#biogerontology)
+ [BIOTOOLS](#biotools)
+ [ECOLOGY](#ecology)
+ [EMR](#emr)
+ [EPIDEMIOLOGY](#epidemiology)
+ [EVOLUTIONARY BIOLOGY](#evolutionary-biology)
+ [GENOMICS](#genomics)
   + [DNA](#dna)
   + [RNA](#rna)
+ [MED-Orthodontics](#med-orthodontics)
+ [MICROBIOLOGY](#microbiology)
   + [Org-biocore](#org-biocore)
+ [MOLECULAR BIOLOGY](#molecular-biology)
+ [NEUROSCIENCE](#neuroscience)
   + [Neural-Networks](#neural-networks)
   + [Neuropsychology](#neuropsychology)
   + [Org-kwikteam](#kwikteam)
+ [Pharma](#pharma)
+ [SYSTEOMICS](#systeomics)
   + [OpenWorm](#openworm)
   + [Synthetic Biology](#synthetic-biology)
   + [Structural Biology](#structural-biology)
+ [Resources-Teaching](#resources-teaching)

----

# [BIOGERONTOLOGY](https://en.wikipedia.org/wiki/Gerontology#Biogerontology)
+ [longevity](https://github.com/jimrybarski/longevity) :: A tool to model global population under various life extension scenarios.
+ [population_pyramid](https://github.com/jimrybarski/population_pyramid) :: Uses matplotlib to create a population pyramid graph.

----

# BIOTOOLS

+ [annotations](https://github.com/tyarkoni/annotations) :: Multimodal stimulus annotation in Python.
+ [antz](https://github.com/jmschrei/antz) :: Bioinformatic tools for biologists, made pythonic!
+ [BioBERT](https://github.com/dmis-lab/biobert) :: a pre-trained biomedical language representation model. https://doi.org/10.1093/bioinformatics/btz682.
+ [bionetwork](https://github.com/machbio/bionetwork) :: Graph Database, a network of everything bio.
+ [Bionitio](https://github.com/bionitio-team/bionitio) :: provides a template for command line bioinformatics tools in various programming languages.
+ [biostar-central](https://github.com/ialbert/biostar-central) :: The software that runs the [Biostars Bioinformatics Q&A](http://www.biostars.org).
+ [Bio_Eutils](https://github.com/jmaupetit/Bio_Eutils) :: The standalone version of the Entrez and Medline BioPython modules.
+ [bipy](https://github.com/roryk/bipy) :: Lightweight bioinformatics pipeline tools using iPython. 
+ [CloudBioLinux](http://cloudbiolinux.org) : configure virtual (or real) machines with tools for biological analyses. [Source Code](https://github.com/chapmanb/cloudbiolinux)
+ [chowda](https://github.com/chrismamil/chowda) :: Python module for analysis of CLAMS (comprehensive lab animal monitoring system) data.
+ [Cosmos](https://cosmos.hms.harvard.edu) :: A Python library for Workflow Management System that tracks massively parallel computing clusters as well as cloud-based services. Download it [here](https://cosmos.hms.harvard.edu/download.html).
+ [CTDopts](https://github.com/genericworkflownodes/CTDopts) :: CTDopts is a module for enabling tools with CTD reading/writing, argument parsing, validating and manipulating capabilities.
+ [Encode-dataframe](https://github.com/daler/encode-dataframe) :: Convert UCSC's ENCODE metadata into pandas DataFrames.
+ [Galaxy](http://galaxyproject.org) is an open, web-based platform for data intensive biomedical research. Use it [online](https://usegalaxy.org/)
   + [bioblend](https://github.com/afgane/bioblend) :: A python library for interacting with CloudMan and Galaxy API.
   + [galaxy_blast](https://github.com/peterjc/galaxy_blast) :: Galaxy wrappers for NCBI BLAST+ and related BLAST tools.
   + [GalaxyConfigGenerator](https://github.com/genericworkflownodes/GalaxyConfigGenerator) :: Given a CTD file, GalaxyConfigGenerator generates a Galaxy wrapper.
   + [galaxy-lib](https://github.com/galaxyproject/galaxy-lib) :: Subset of Galaxy (http://galaxyproject.org/) core code designed to be used as a library.
+ [rabix](https://github.com/rabix/rabix) :: Reproducible Analyses for Bioinformatics.
+ [samtools-trio-nexus](https://github.com/bpow/samtools-trio-nexus) :: An applet for running samtools on trios (child and both parents) on DNAnexus.
+ [tiny-test-data](https://github.com/roryk/tiny-test-data) :: Super small biological datasets for unit testing. 

## Docker
+ [biopython_docker](https://github.com/biopython/biopython_docker) :: Dockerfiles to build Biopython instances.


----
  
# EMR
+ [AuShadha](https://github.com/dreaswar/AuShadha) :: AuShadha (औषध) means medicine in Sanskrit. This is a Electronic Medical Records (EMR) and Public Health Management for small clinics written in Django and Dojo.

----

# ECOLOGY
+ [Macroeco](http://macroeco.org) :: Ecological pattern analysis in Python. Get the [Source code](https://github.com/jkitzes/macroeco) and [Documentation](http://macroeco.org/en/latest/).
+ [Bird-colony](https://github.com/melizalab/django-bird-colony) :: A Django application used to manage bird colonies (including breeding colonies).
+ [Python-ecology](http://datacarpentry.github.io/python-ecology) :: Data Carpentry Python lessons on ecology. Source [code](https://github.com/datacarpentry/python-ecology) on github. 

### Bioacoustics
+ [Chirp](http://melizalab.github.io/chirp/) :: A set of related tools for pitch-based analysis and comparison of bioacoustic signals. [Source code](https://github.com/melizalab/chirp)

----

# EPIDEMIOLOGY
+ [Epibayes](https://github.com/jzelner/epibayes) :: (Rudimentary) tools for epidemiological modeling w/Bayesian statistics. 
+ [Epipy](https://github.com/cmrivers/epipy) :: Epipy is a Python package for epidemiology.
+ [episounds](https://github.com/cmrivers/episounds) :: A sonification of human to human transmission of Middle East Respiratory Syndrome Coronavirus. 
+ [Epitopes](https://github.com/hammerlab/epitopes) → Python interface to immunology and bioinformatics datasets (i.e. IEDB, cancer antigens, TCGA mutant peptides).

----

# [EVOLUTIONARY BIOLOGY](https://en.wikipedia.org/wiki/Category:Evolutionary_biology)
+ [DEAP](https://github.com/DEAP/deap) :: Distributed Evolutionary Algorithms in Python. [Documentation](http://deap.readthedocs.org/)
   + [Notebooks](https://github.com/DEAP/notebooks) on how to use Distributed Evolutionary Algorithm in Python (DEAP). 
+ Genetic Programming → [Evolution of Mona Lisa](http://rogeralsing.com/2008/12/07/genetic-programming-evolution-of-mona-lisa/). Here is the [non-Python LGPL licensed source code](http://code.google.com/p/alsing/downloads/list), which should make an interesting project to port to Python. 
   + [Pierre Lindenbaum's implementation of the Genetic Algorithm described by Roger Alsing](http://plindenbaum.blogspot.in/2008/12/random-notes-2008-12.html) above.
+ [parallyze](https://github.com/rohanmaddamsetti/parallyze) :: Software to analyze parallel genome evolution.
+ [phylowgs](https://github.com/morrislab/phylowgs) :: Application for inferring subclonal composition and evolution from whole-genome sequencing data. 
+ [screed](https://github.com/ged-lab/screed) (GED@MSU.EDU) :: A simple read-only sequence database, designed for short reads from the Genomics, Evolution, and Development Lab. [Documentation](http://readthedocs.org/docs/screed/en/latest/).
+ [singlecell](https://github.com/roryk/singlecell) :: single-cell data analysis incubator. 

----

# GENOMICS
+ [allbiotc2](https://github.com/ALLBio/allbiotc2) :: Benchmark pipeline for Structural Variation analyses, funded by the [ALLBio](http://www.allbioinformatics.eu/doku.php?id=start).
+ [anhima](https://github.com/alimanfoo/anhima) :: Analyse genetic variation
+ [ASAP](https://github.com/ddofer/asap) :: Amino-Acid Sequence Annotation Predictor. 
+ [bamslider](https://github.com/vsbuffalo/bamslider) :: Sliding windows in BAM/SAM files with Python's deques.
+ [Bamsurgeon](https://github.com/adamewing/bamsurgeon) :: Tools for adding mutations to existing .bam files, used for testing mutation callers.
+ [batch_clustalo](https://github.com/koshlan/batch_clustalo) :: Multiple Sequences Alignments in Batch.
+ [bcbio-nextgen](https://github.com/chapmanb/bcbio-nextgen) is validated, scalable, community developed variant calling and RNA-seq analysis. [Documentation](https://bcbio-nextgen.readthedocs.org)
   + [bcbio-test-data](https://github.com/roryk/bcbio-test-data) :: test data for bcbio-nextgen.
   + [bcbio-conda](https://conda.binstar.org/bcbio) :: Conda recipes for bcbio python code and dependencies. [Source Code](https://github.com/chapmanb/bcbio-conda)
   + [bcbio.coverage](https://github.com/lpantano/bcbio.coverage) :: utils functions for exome coverage plots as quality controls.
+ [biomartpy](https://github.com/daler/biomartpy) :: Simple interface to BioMart (Python -> rpy2 -> R/BioConductor's biomaRt).
+ [BioSeq](https://github.com/MeHelmy/BioSeq) is a python lib for Sequence Alignment Map (SAM), a standard data storage format for DNA sequencing.
+ [brat](https://github.com/nlplab/brat) :: Brat rapid annotation tool - for all your textual annotation needs http://brat.nlplab.org
+ [BreakSeq2](https://github.com/bioinform/breakseq2) :: Ultrafast and accurate nucleotide-resolution analysis of structural variants.
+ [Chanjo](http://www.chanjo.co/) :: This package provides a better way to analyze coverage data in clinical sequencing. [Source Code](https://github.com/robinandeer/chanjo). 
+ [chrom_sweep](https://github.com/arq5x/chrom_sweep) :: Sweep-line algorithm for genomic features. Detect overlaps on large files w/ minimal memory.
+ [codachrom](https://github.com/arq5x/codachrom) :: Chromosomal copy number tools.
+ [CompleteGenomicsTools](https://github.com/seandavi/CompleteGenomicsTools) →  Complete Genomics provides whole-genome sequencing using DNA nanoball arrayed sequencing. Software for manipulating and visualizing Complete Genomics data, with a [focus on cancer](http://watson.nci.nih.gov/~sdavis/software/cgent/)
+ [CaPSID](https://pypi.python.org/pypi/capsid/1.4.3) (Computational Pathogen Sequence IDentification) :: A comprehensive open source platform which integrates a high-performance computational pipeline for pathogen sequence identification and characterization in human genomes and transcriptomes together with a scalable results database and a user-friendly web-based software application for managing, querying and visualizing results. [Source code](https://github.com/capsid/capsid) and the [Documentation Wiki](https://github.com/capsid/capsid/wiki/).
+ [cyvcf](https://github.com/arq5x/cyvcf) :: A fast Python library for VCF files leveraging Cython for speed. 
+ [collage-dicty](https://github.com/marinkaz/collage-dicty) :: Gene prioritization by compressive data fusion and chaining. [Paper](http://dx.doi.org/10.1371/journal.pcbi.1004552).
+ [ETE - Environment for Tree Exploration](http://etetoolkit.org/treeview/) :: A python programming toolkit that assists in the automated manipulation, analysis and visualization of any type of hierarchical trees. This includes phylogenetic trees, clustering results and profile-based trees. It supports node annotation, programatic tree drawing, circular visualization, SVG, PNG and PDF image rendering, and more! View the [source code](https://github.com/jhcepas/ete) on github. 
+ [figmop](https://github.com/dave-the-scientist/figmop) :: Finding Genes using Motif Patterns. 
+ [fusenet](https://github.com/marinkaz/fusenet) :: Gene network inference by fusing data from diverse distributions. [Paper](http://bioinformatics.oxfordjournals.org/content/31/12/i230).
+ [Gemini](https://github.com/arq5x/gemini) :: A lightweight db framework for disease and population genetics.
+ [genomeEdit](https://github.com/khaikhai/genomeEdit) :: Python script to count SMRT sequenced reads of amplicons generated from genome editing using CRISPR or TALEN methods. 
+ [gffutils](https://github.com/daler/gffutils) is a Python package for working with and manipulating the GFF and GTF format files typically used for genomic annotations. [Documentation](http://pythonhosted.org/gffutils/).
+ [gtf-parse-off](https://github.com/dcjones/gtf-parse-off) :: Experiments with parsing gene transfer format (GTF).
+ [gvl_flavor](https://github.com/afgane/gvl_flavor) :: Genomics Virtual Lab (GVL) flavor for CloudBioLinux.
+ [g-quad](https://github.com/roryk/g-quad) :: Calculates the G-quadraplex score for a FASTA file of sequences.
+ [Harvest](https://github.com/marbl/harvest) is a suite of core-genome alignment and visualization tools for quickly analyzing thousands of intraspecific microbial genomes. It includes Parsnp, a fast core-genome multi-aligner, and Gingr, a dynamic visual platform. Combined they provide interactive core-genome alignments, variant calls, recombination detection, and phylogenetic trees.
+ [HasBin](https://github.com/bpow/HasBin) :: A project for keeping track of genes/diagnoses and custom annotations.
+ [hapmuc](https://github.com/usuyama/hapmuc) ::  A somatic mutation caller, which can utilize the information of heterozygous germline variants near candidate mutations. 
+ [hts-python](https://github.com/brentp/hts-python) :: pythonic wrapper for libhts.
+ [HTSeq](https://pypi.python.org/pypi/HTSeq) :: A framework to process and analyze data from high-throughput sequencing (HTS) assays. [Documentation](http://www-huber.embl.de/users/anders/HTSeq/doc/index.html).
+ [khmer](https://github.com/ged-lab/khmer) :: In-memory nucleotide sequence k-mer counting, filtering, graph traversal and more. [Documentation](http://khmer.readthedocs.org)
+ [KGML](https://github.com/widdowquinn/KGML) :: Parse, manipulate, download, and visualise KGML (KEGG markup language) biological pathway data.
+ [KneadData](https://bitbucket.org/biobakery/kneaddata) :: A tool designed to perform quality control on metagenomic sequencing data, especially data from microbiome experiments.
+ [MAF](https://github.com/vsbuffalo/maf) :: a light framework to pipeline short read mapper/aligner testing.
+ [metaseq](https://github.com/daler/metaseq) :: A Framework for integrated analysis and plotting of ChIP/RIP/RNA/*-seq data. [Documentation](http://packages.python.org/metaseq/).
+ [methtools](https://github.com/bgruening/methtools) :: Tools for the processing of genome-wide bisulfite sequencing data.
+ [MetaSVMerge](https://github.com/bioinform/metasv) :: An accurate method-aware merging algorithm for structural variations.
+ [mmgenome](http://madsalbertsen.github.io/mmgenome/) :: Tools for extracting individual genomes from metagneomes.
+ [MixClone](https://github.com/uci-cbcl/MixClone) :: A mixture model for inferring tumor subclonal populations.
+ [nanopore-scripts](https://github.com/arq5x/nanopore-scripts) :: Various scripts and recipes for working with nanopore data. 
+ [nexus-fusetester](https://github.com/bpow/nexus-fusetester):: A simple applet to test use of python-llfuse to access files within a DNAnexus node.
+ [Oncotator](https://github.com/ankurdave/oncotator) :: A web application for annotating human genomic point mutations and indels with data relevant to cancer researchers. See, http://www.broadinstitute.org/oncotator
+ [PAAP](https://github.com/RILAB/paap) :: Preprocessing and Alignment Pipeline (PAAP) using Slurm array jobs.
+ [peddy](https://github.com/brentp/peddy) :: An API for dealing with pedigree files.
+ [piper](https://github.com/NationalGenomicsInfrastructure/piper) :: A genomics pipeline build on top of the GATK Queue framework. 
+ [Platypus](https://github.com/andyrimmer/Platypus) :: The Platypus variant caller.
+ [poretools](https://github.com/arq5x/poretools) :: A toolkit for working with Oxford nanopore data. 
+ [PyBEDtools](https://github.com/daler/pybedtools) is a Python wrapper for Aaron Quinlan's [BEDtools programs](https://github.com/arq5x/bedtools), which are widely used for genomic interval manipulation or "genome algebra". Pybedtools extends BEDTools by offering feature-level manipulations from with Python. See the [online documentation, including installation instructions](http://pythonhosted.org/pybedtools/).
+ [PyVCF](https://github.com/jamescasbon/PyVCF) :: A Variant Call Format reader for Python. [Documentation](http://pyvcf.readthedocs.org/en/latest/index.html).
+ [pygenetorrent](https://github.com/hammer/pygenetorrent) → A Python client for GeneTorrent.
+ [rbac-23andme-oauth2](https://github.com/offapi/rbac-23andme-oauth2) :: Genetic Access Control using the 23andME API.
+ [rubra](https://github.com/bjpop/rubra) :: Infrastructure code to support DNA pipeline.
+ [scurgen](https://github.com/arq5x/scurgen) :: A tool for detecting patterns in genomic data with space filling curves.
+ [seq2seq-attn](https://github.com/harvardnlp/seq2seq-attn) :: Sequence-to-sequence model with LSTM encoder/decoders and attention. http://nlp.seas.harvard.edu/code
+ [seqscripts](https://github.com/roryk/seqscripts) :: Scripts for parsing the output from the Tophat/Cufflinks/Cuffcompare pipeline into other formats and analyzing and editing GTF files. 
+ [singlecell](https://github.com/roryk/singlecell) data analysis incubator for analysing single-cell data generated by the method outlined here: http://biorxiv.org/content/early/2014/03/05/003236
+ [Smash](https://github.com/amplab/smash) :: A benchmarking toolkit for variant calling.
+ [theprimerdirective](https://github.com/jimrybarski/theprimerdirective) :: A Python interface to Primer3. 


### DNA
+ [dna-traits](https://github.com/cslarsen/dna-traits) :: A fast 23andMe genome text file parser.
+ [PyPore](https://github.com/jmschrei/PyPore) :: This package is based off of a few core data analysis packages in order to provide a consistent and easy framework for handling nanopore data in the UCSC nanopore lab.
+ [Railroadtracks](https://github.com/Novartis/railroadtracks) :: A Python package to handle connected computation steps for DNA and RNA Seq.

### RNA
+ [findorf](https://github.com/vsbuffalo/findorf) :: ORF prediction of de novo transcriptome assemblies and contig annotation tool designed to be non-model organism-friendly.
+ [flotilla](https://github.com/YeoLab/flotilla) :: A Python package for visualizing transcriptome (RNA expression) data from hundreds of samples - Reproducible machine learning analysis of gene expression and alternative splicing data. [Documentation](http://yeolab.github.io/flotilla/docs).
+ [gimme](https://github.com/likit/gimme) :: A lightweight reference-guided Alignment-based assembler for transcriptome analysis.
+ [RNASeqReadSimulator](http://www.cs.ucr.edu/~liw/rnaseqreadsimulator.html) :: A simple tool to generate simulated single-end or paired-end RNA-Seq reads. [Source Code](https://github.com/davidliwei/RNASeqReadSimulator).
+ [tcgaparse](https://github.com/machbio/tcgaparse) :: Python Scripts to Parse TCGA data.
+ Wikipedia's [list of RNA-Seq bioinformatics tools](http://en.wikipedia.org/wiki/List_of_RNA-Seq_bioinformatics_tools), not all of which are in Python, but they may have an API one can use.
+ [YAP](https://github.com/Novartis/yap) :: An extensible parallel framework, written in python using openmpi libraries that allows researchers to quickly build high throughput big data pipelines without extensive knowledge of parallel programming.

###### Research Publications
+ Pieter Kelchtermans, Wout Bittremieux, Kurt De Grave, Sven Degroeve1, Jan Ramon, Kris Laukens, Dirk Valkenborg, Harald Barsnes, Lennart Martens; [Machine learning applications in proteomics research: How the past can boost the future](http://onlinelibrary.wiley.com/doi/10.1002/pmic.201300289/full);21JAN2014
+ Kenneth Verheggen, Harald Barsnes, Lennart Martens, [Distributed computing and data storage in proteomics: Many hands make light work, and a stronger memory](http://onlinelibrary.wiley.com/doi/10.1002/pmic.201300288/full), 27NOV2013.
+ Elien Vandermarliere, Lennart Martens; [Protein structure as a means to triage proposed PTM sites](http://onlinelibrary.wiley.com/doi/10.1002/pmic.201200232/full); 04FEB2013.
+ Marc Vaudel, Albert Sickmann and Lennart Martens; [Current methods for global proteome identification](http://informahealthcare.com/doi/abs/10.1586/epr.12.51); OCT2012.

###### Resources
__Jupyter notebooks/Cookbooks, tutorials and learning materials from Workshops, hackathon codebases, etc..__
+ Notes: [An introductory approach (with an application to the Prisoner's Dilemma) to Genetic Algorithms](http://www2.econ.iastate.edu/tesfatsi/holland.gaintro.htm)
+ [Benchmarking HDF5 and BLZ for genotype data storage and access](http://nbviewer.ipython.org/gist/alimanfoo/67fdcf58e364763fd0b6/benchmark_hdf5_blz.ipynb)
+ [2013-norwich-bioinfo](https://github.com/ngs-docs/2013-norwich-bioinfo) :: 2013 advanced bioinformatics workshop at Norwich.
+ [2013-norwich-biology](https://github.com/ngs-docs/2013-norwich-biology) :: Advanced biology workshop at Norwich.
+ [2013-davis-assembly](https://github.com/ngs-docs/2013-davis-assembly)
+ [2015-khmer-wok4-multimap](https://github.com/ctb/2015-khmer-wok4-multimap).
+ [2015-python-intro](https://github.com/ngs-docs/2015-python-intro) :: An Introduction to Python.

----

# MED-[Orthodontics](https://en.wikipedia.org/wiki/Orthodontics)
+ [Q3DCExtension](https://github.com/DCBIA-OrthoLab/Q3DCExtension) :: Compute 2D angles using placed fiducials.

----

# MICROBIOLOGY
+ [Ebola](https://github.com/cmrivers/ebola) :: Data for the 2014 Global Ebola outbeak.
+ [fylm](https://github.com/jimrybarski/fylm) :: Extracts data from Fission Yeast Lifespan Microdissector images.
+ [molecbio](https://github.com/dave-the-scientist/molecbio) :: Molecular biology informatics functions. 
+ [patternHmm](https://github.com/dave-the-scientist/patternHmm) :: Matching patterns of symbols using a profile Hidden Markov Model.
+ [phageParser](https://github.com/BillMills/phageParser) :: A parser to extract the relevant data from the [PhagesDB](http://phagesdb.org/) Mycobacteriophage database.
+ [ProkaryMetrics](https://github.com/smdabdoub/ProkaryMetrics) :: Visualize and analyze 3D biofilm data from fluorescent micrographs. http://justicelab.org/pkm


## Org-[biocore](https://github.com/biocore)
+ [burrito](https://github.com/biocore/burrito) :: Python framework for controlling command-line applications.
+ [burrito-fillings](https://github.com/biocore/burrito-fillings) :: Application controllers for command line bioinformatics applications 
+ [emperor](https://github.com/biocore/emperor) :: Emperor a tool for the analysis and visualization of large microbial ecology datasets. http://emperor.colorado.edu
+ [glowing-dangerzone](https://github.com/biocore/glowing-dangerzone) :: Easy SQL connection handlers.
+ [QIIME](http://www.qiime.org) :: Quantitative Insights Into Microbial Ecology. [Official repository](https://github.com/biocore/qiime) for software and unit tests.
   __Fork__
   + [phylotoast](https://github.com/smdabdoub/phylotoast) :: Additions to the QIIME (Quantitative Insights Into Microbial Ecology) project including data analysis, visualizations, and cluster-computing support. DOCS :: http://qiime-tools.readthedocs.org
+ [mustached-octo-ironman](https://github.com/biocore/mustached-octo-ironman) :: Easy dispatched compute via in a Tornado environment.
+ [my-microbes](https://github.com/biocore/my-microbes) :: A set of tools for delivering personal microbiome results to individuals participating in microbiome sequencing studies. 
+ [PyNAST](http://biocore.github.io/pynast/) :: Python Nearest Alignment Space Termination tool - Official repository for software and unit tests.
+ [qiita](https://github.com/biocore/qiita) :: A QIIME databasing effort. 
+ [Platypus Conquistador](https://github.com/biocore/Platypus-Conquistador) :: A bioinformatic command line tool for the confirmation of the presence/absence of a specific taxon (or set of taxa) in environmental shotgun sequence reads. 
+ [pyqi](https://github.com/biocore/pyqi) :: Tools for developing and testing command line interfaces in Python. 
+ [scikit-bio](http://scikit-bio.org) :: Core objects, functions and statistics for working with biological data in Python. [Source code](https://github.com/biocore/scikit-bio).
+ [StrainDB](https://github.com/biocore/StrainDB) :: A database to store strain genomes.
+ [tax2tree](https://github.com/biocore/tax2tree) :: Automated taxonomy decoration onto a tree.
+ [verman](https://github.com/biocore/verman) :: Python module version information.

----

# MOLECULAR BIOLOGY
+ [BioPython](http://biopython.org) :: project is an international association of developers of freely available Python tools for computational molecular biology. [Source code](https://github.com/biopython/biopython)
+ [bcbb](https://github.com/chapmanb/bcbb) :: An incubator (collection) of useful bioinformatics code related to biological analysis, primarily in Python and R. [Blog](http://bcbio.wordpress.com)
+ [mdtraj](http://www.mdtraj.org) :: A modern, open library for the analysis of molecular dynamics trajectories. [Source code](https://github.com/mdtraj/mdtraj) on GHub.
+ [MMTK](http://dirac.cnrs-orleans.fr/MMTK/) :: The Molecular Modelling Toolkit is an Open Source program library for molecular simulation applications.
+ [MSMBuilder](http://msmbuilder.org/) :: An open source software package for automating the construction and analysis of Markov State Models for Biomolecular Conformational Dynamics. MSMs are a powerful means of modeling the structure and dynamics of molecular systems, like proteins. Installation is easy via Anaconda : [read the documentation](http://msmbuilder.org/installation.html), or the [source code](https://github.com/pandegroup/msmbuilder) to report bugs and contribute patches.
+ [tRNA_evo](https://github.com/vhsvhs/trna_evo) :: This project analyzes the evolution of tRNAs, from across the tree of life. How prevalent is anticodon switching?

###### Resources
__Jupyter notebooks/Cookbooks, tutorials and learning materials from Workshops, hackathon codebases, etc..__
+ [GFF Parsing](http://biopython.org/wiki/GFF_Parsing) notes.

----

# NEUROSCIENCE
+ [BrainImagingPipelines](https://github.com/INCF/BrainImagingPipelines) → Optimized Nipype pipelines for brain imaging.
+ [brainx](https://github.com/nipy/brainx) :: Tools for analysis of brain imaging-derived networks, based on NetworkX.
+ [CodeNeuro](http://codeneuro.org/) :: Bringing neuroscience and data science together.
+ [CogSlots](https://github.com/cbattista/CogSlots) :: Simulated Slot Machine for the behavioral sciences.
+ [CPAC](https://github.com/FCP-INDI/C-PAC) :: A configurable pipeline for rsfMRI/connectome analyses.
+ [DCMPy](https://github.com/armaneshaghi/DCMPy) :: A Python module for longitudinal surface-based DCM of fMRI data.
+ [dicomsort](https://github.com/pieper/dicomsort) :: A project to provide custom sorting and renaming of dicom files.
+ [dipy](http://dipy.org) → Diffusion MR Imaging in Python. [Source code](https://github.com/nipy/dipy)
+ [ERPy](https://github.com/nejstastnejsistene/ERPy) :: An open-source neuroscience application for visualization and analysis of electroencephalographic data using the Python programming language.
+ [fit_neuron](https://github.com/nicodjimenez/fit_neuron) :: A neuroscience python package for the estimation and evaluation of neural models from patch clamp neural recordings, including a library of spike distance metrics.
+ [fmri-analysis-vm](https://github.com/poldrack/fmri-analysis-vm) :: A VM setup for use in fMRI analysis and education.
+ [ICA-AROMA](https://github.com/rhr-pruim/ICA-AROMA) :: Software package of ICA-AROMA; a data-driven method to identify and remove motion-related independent components from functional MRI data.
+ [MIEN](http://mien.msu.montana.edu/) :: Python tool chain framework and data visualization suite for neuroscience. [Source code](https://github.com/gic888/MIEN)
+ [MNE](http://martinos.org/mne/) :: Magnetoencephalography (MEG) and Electroencephalography (EEG) in Python. [Source code](https://github.com/mne-tools/mne-python)
   + [Basic introduction to MNE Python processing](https://github.com/mne-tools/mne-python-intro).
+ [nengo](https://github.com/nengo/nengo) :: A Python library for creating and simulating large-scale brain mode.
+ [NeuroDebian](http://neuro.debian.net). [Source code](https://github.com/neurodebian)
+ [Neurosciences](https://github.com/rougier/Neurosciences) :: Computational Neurosciences repository.
+ [NIPY](https://github.com/nipy) developers group on GH.
+ [nibabel](http://nipy.org/nibabel/) :: Python package to access a cacophony of neuro-imaging file formats. [Source code](https://github.com/nipy/nibabel).
+ [NIDM](https://github.com/incf-nidash/nidm) ::  Neuroimaging Data Model (NIDM) describing neuroimaging data and provenance. http://nidm.nidash.org
   + [nidm-training](https://github.com/incf-nidash/nidm-training) :: Training materials for learning NIDM.
+ [nipype](https://github.com/nipy/nipype) :: Workflows and interfaces for neuroimaging packages with a [nightly builds](http://www.mit.edu/~satra/nipype-nightly/).
+ [nilearn](http://nilearn.github.io) is a machine learning tool for NeuroImaging in Python. [Source code](https://github.com/nilearn/nilearn)
   + [NiLearn course](https://github.com/GaelVaroquaux/nilearn_course) :: Functional connectivity with NiLearn.
+ [Nitime](http://nipy.org/nitime) :: Timeseries analysis for neuroscience data.
+ [pipelines](https://github.com/NeuroanatomyAndConnectivity/pipelines) :: Neuroimgaing data processing pipelines used in the lab.
+ [PredPy](https://github.com/armaneshaghi/PredPy) :: is a collection of IPython notebooks predicting multiple sclerosis functional composite (MSFC) disability from MRI scans in people with MS.
+ [PyBrain](http://pybrain.org) and its [installation](https://github.com/pybrain/pybrain/wiki/installation)
+ [PyCogMo](https://github.com/agravier/pycogmo) :: is a modular Python framework to develop computational experiments in Cognitive Neuroscience. It makes use of [PyNN](http://neuralensemble.org) and adds task-level scheduling and facilities (learning and testing), and visualisation functions.
+ [pycone](https://github.com/bgalbraith/pycone) :: Python in Computational Neuroscience
+ [pydcemri](https://github.com/welcheb/pydcemri) :: Python module for processing dynamic contrast enhanced magnetic resonance imaging (DCE-MRI) data. Given a T1-weighted, dynamic, contrast-enhanced data set, a multiflip data set, and either an AIF or blood curves, produce maps of Ktrans, ve, and vp.
+ [Pycortex](http://pycortex.org) :: A python-based toolkit for surface visualization in fMRI data. [Source Code](https://github.com/gallantlab/pycortex) on Github.
+ [pydicom](https://github.com/darcymason/pydicom) :: Read, modify and write DICOM files with python code.
+ [PYEZMINC](https://github.com/BIC-MNI/pyezminc) :: is a python module to read and write MINC files.
+ [PyMVPA](http://www.pymvpa.org) → MultiVariate Pattern Analysis in Python. [Source code](https://github.com/PyMVPA/PyMVPA)
+ [pyneurovault](https://github.com/NeuroVault/pyneurovault) :: A python wrapper for the NeuroVault API (in dev) with [documentation](http://www.neurovault.org/api-docs)
+ [PySurfer](https://github.com/nipy/PySurfer) :: Cortical neuroimaging visualization in Python. [Online documentation (stable)](http://pysurfer.github.com/) and [Mailing list](http://mail.scipy.org/mailman/listinfo/nipy-devel)
+ [PyView](https://github.com/lrajmohan/PyView) :: A project written in Python to perform experiments on learning and decision making used in the department of Neuroscience at Stony Brook University.


### Neural Networks
+ [biaxial-rnn-music-composition](https://github.com/hexahedria/biaxial-rnn-music-composition) :: A recurrent neural network designed to generate classical music. 
+ [blocks](https://github.com/mila-udem/blocks) :: A Theano framework for building and training neural networks.
+ [cnn-vis](https://github.com/jcjohnson/cnn-vis) :: Use CNNs to generate images.
https://github.com/hexahedria/biaxial-rnn-music-composition :: A recurrent neural network designed to generate classical music. 
+ [Kayak](https://github.com/HIPS/Kayak) :: a library for automatic differentiation with applications to deep neural networks. 
+ [nupic](https://github.com/numenta/nupic) :: a Platform for Intelligent Computing, brain-inspired machine intelligence platform, and biologically accurate neural network based on cortical learning algorithms by [Numenta](http://numenta.org/).
+ [nupic.studio](https://github.com/nupic-community/nupic.studio) :: NuPIC Studio is a powerful all­-in-­one tool that allows users create a HTM neural network from scratch, train it, collect statistics, and share it among the members of the community.
+ [neurobank](https://github.com/melizalab/neurobank) :: A simple, low-overhead data management system designed for neural and behavioral data, but could be used for other kinds of experiments.
+ [optofit](https://github.com/HIPS/optofit) :: A python framework for fitting biophysical models to optically recorded neural signals. 
+ [remote_associates_test](https://github.com/ikajic/remote_associates_test) :: Neural network simulation of the Remote Associates Test.
+ [thunder](http://thunder-project.org/thunder/docs/) :: neural data analysis in spark.

###### Resources
+ [Practical-neuroimaging](http://practical-neuroimaging.github.io) :: Exercises for 2015 Berkeley practical neuroimaging class. [Source code](https://github.com/practical-neuroimaging/pna2015).
+ [deepdream](https://github.com/google/deepdream) :: This repository contains IPython Notebook with sample code, complementing Google Research blog post about Neural Network art.
+ [CodeNeuro-notebooks](https://github.com/CodeNeuro/notebooks) :: Interactive notebooks for trying analyses and exploring datasets. [Website](http://notebooks.codeneuro.org).
__Jupyter notebooks/Cookbooks, tutorials and learning materials from Workshops, hackathon codebases, etc..__
+ _Open book_ on [Neural Networks and Deep Learning](http://neuralnetworksanddeeplearning.com/)
   * Chapter-1. [Using neural nets to recognize handwritten digits](http://neuralnetworksanddeeplearning.com/chap1.html)
   * Chapter-2. [How the backpropagation algorithm works](http://neuralnetworksanddeeplearning.com/chap2.html)
   * Chapter-3. [Improving the way neural networks learn](neuralnetworksanddeeplearning.com/chap3.html)
   * Chapter-4. Neural nets: the big picture
   * Chapter-5. Deep learning
   * Chapter-6. The state of the art in image recognition
   * Chapter-7. The future of neural nets
+ [Deep Neural Networks: A Getting Started Tutorial](http://www.visualstudiomagazine.com/articles/2014/06/01/deep-neural-networks.aspx)
+ [Simple neural net setup practice](https://github.com/nyghtowl/Neural_Net_Practice)
+ [Python in NeuroImaging trifold](http://www.onerussian.com/tmp/nipy-handout.pdf)
+ [Python in Electrophysiology trifold](http://www.onerussian.com/tmp/eppy-handout.pdf)
+ [python-for-neuroscience](https://github.com/kylerbrown/python-for-neuroscience) :: A workshop for teaching python to neuroscientists with some previous programming experience.

### Neuropsychology
+ [OpenPsyc](https://github.com/cbattista/OpenPsyc) :: Open source Python scripts for Psychology and the Neurosciences.
+ [posner](https://github.com/psychopy/posner) :: The Posner task as a demo for workshop.
+ [PsychoPy](http://www.psychopy.org) :: An open-source package for creating psychology stimuli in Python. PsychoPy combines the graphical strengths of OpenGL with the easy Python syntax to give psychophysics a free and simple stimulus presentation and control package. [Source code](https://github.com/psychopy/psychopy)
+ [python-pyepl](http://neuro.debian.net/pkgs/python-pyepl.html) :: A module for coding psychology experiments in Python.
+ [VisionEgg](http://www.visionegg.org) :: is another open-source package for creating psychology stimuli in Python, with a specific emphasis on visual stimuli. [Source code](https://github.com/visionegg/visionegg).

### Org-[kwikteam](https://github.com/kwikteam)
__Large-scale electrophysiological data analysis framework in Python.__
+ [phy](https://github.com/kwikteam/phy) :: Interactive electrophysiological data analysis package.

----

# Pharma
+ [ops-search](https://github.com/openphacts/ops-search) :: Open PHACTS search service. 
 
----

# [SYSTEOMICS](http://en.wikipedia.org/wiki/Systems_biology)

## OpenWorm
+ The [OpenWorm](http://www.openworm.org) project aims to build the first comprehensive computational model of the Caenorhabditis elegans (C. elegans), a microscopic roundworm. Read more on [Wikipedia](http://en.wikipedia.org/wiki/OpenWorm) and their various [projects on github](https://github.com/openworm/).
+ [PyOpenWorm](https://github.com/openworm/PyOpenWorm) :: Unified, simple data access library for data & facts about c. elegans anatomy.
+ [org.geppetto](https://github.com/openworm/org.geppetto) :: [Geppetto](http://geppetto.org) is a web-based multi-algorithm, multi-scale Systems Biology simulation platform engineered to support the simulation of complex biological systems and their surrounding environment. 

## Synthetic Biology
+ [SynBio](https://bitbucket.org/chapmanb/synbio/src/) :: is a Python Synthetic Biology library collection of synthetic biology code.

## Structural Biology
+ [CheShift](https://github.com/aloctavodia/cheshift) is a software for prediction of 13Cα and 13Cβ chemical shifts and validation of protein structures.
+ [Halpain-Lab](http://www.ccb.tu-dortmund.de/groups/CB/bastiaens/dehmelt/NeuriteQuant/) :: Automated analysis for Fluorescent Microscopy of iPSC and in-vitro cell culture images. [Source Code](https://github.com/GrantRVD/Halpain-Lab).

###### Resources
__Jupyter notebooks/Cookbooks, tutorials and learning materials from Workshops, hackathon codebases, etc..__
+ [SBioA](https://github.com/aloctavodia/SBioA) :: An introduction to Structural Bioinformatics Algorithms and scientific computing using Python/PyMOL.
+ [PymolWiki](http://www.pymolwiki.org/index.php/Main_Page) :: Guide for PyMol users.

----

# Resources-Teaching
__NotaBene__:: Some resources and teaching aids listed here are not Python language specific but you may be able to find something common and useful that can be reused and shared with attribution if it is released under a CC-license.

+ [Python for biologists](http://www.pythonforbiologists.com) website.
+ [Common Bioinformatics Teaching Resources](http://www.embl.de/~seqanal/courses/commonCourseContent/indexCommonCourseContent.html) by Aidan Budd at EMBL.
+ A google document listing [Python for biologists resources](https://docs.google.com/spreadsheets/d/1BjKsN0B1hqd4dJW5slZ5KPuToCjSMRyA7Bl8MwWrbS4/edit#gid=0) by Lenny Teytelman (@lteytelman on twitter). 
+ [Software Carpentry](https://software-carpentry.org/lessons.html) teaching material.
+ [Train online](https://www.ebi.ac.uk/training/online/) provides free courses on Europe's most widely used data resources, created by experts at EMBL-EBI and collaborating institutes.
+ [TrainingPhyloIntro](https://github.com/aidanbudd/trainingPhyloIntro) :: Practical exercises and slides for teaching computational molecular evolution.
+ [capetown2014](https://github.com/aidanbudd/capetown2014) :: Teaching material for EMBO Practical Course on Computational analysis of protein-protein interactions - From sequences to networks run in Cape Town 2014.
+ [course_EMBO_at_TGAC_PPI_Sep2015](https://github.com/aidanbudd/course_EMBO_at_TGAC_PPI_Sep2015) :: Program and some material for EMBO Practical Course on computational analysis of protein-protein interactions: From sequences to networks, taking place at TGAC, Norwich, UK, Sep-Oct 2015.
+ Ian Quigley's (@mucociliary on twitter) notes on teaching [applied bioinformatics / compbio](http://daudin.icmb.utexas.edu/wiki/index.php/List_of_sessions).
+ [Ubiquitous Genomics](http://ubiquitousgenomics.teamerlich.org) :: COMSE6998/ Class 2015 by Prof. Yaniv Erlich. The [source is available on github](https://github.com/erlichya/ubiquitousgenomics).
+ [gradient analysis](https://github.com/NeuroanatomyAndConnectivity/gradient_analysis) :: Notebooks on Gradient analysis.


## Other Resources
+ [NIH project reporter](http://projectreporter.nih.gov/reporter.cfm)
+ [ExPASy](http://expasy.org/) :: The SIB Bioinformatics Resource Portal which provides access to scientific databases and software tools (i.e., resources) in different areas of life sciences including proteomics, genomics, phylogeny, systems biology, population genetics, transcriptomics etc.
+ [MSDS](http://www.msdsonline.com/msds-search/)
+ [Open WetWare](http://openwetware.org/wiki/Main_Page) :: A wiki for molecular biology protocols.
+ [MolBio Tools](http://molbiol-tools.ca) :: Tools for molecular biology.
+ [Protocol-online](http://www.protocol-online.org) :: A Q&A portal for molecular biology protocols.

----
